### DISCUSSION TITLE
use BoB on first row

### DISCUSSION THREAD

**Discussion by markusschloesser** [markusschloesser/MackieC4_P3#71] [Ideas]
Live has a BoB (Best of Banks) bank for the most important 8 DeviceParams, which we currently do not use.
We might though introduce it in row1_encoders IN ADDITION to having the other banks coming afterwards.
Might be good to have cutoff etc there. 
Thoughts?

**Comment by markusschloesser** [comment]
"If you put parameters only in the BoB bank then they will not show up" need to be in both, normal bank and BoB

**Reply by BilldarBagdar** [reply]
> If you put parameters only in the BoB bank then they will not show up

Not sure what you mean, but it may just be the way you tried to implement.  Theoretically, we should be able to map "device parameters" to encoders in any order we want.  As long as there is at least 1 parameter to map, it could be mapped to any single encoder up through all 32 encoders.

This seems like a great idea for an Alt + Channel Strip mode, and/or what you see when you hold Shift in Channel mode.  I mean  regular "channel strip" mode now shows the "channel banks" of devices (names) loaded on the track so you can button-click jump to device mode of that device regardless which device is "currently selected" on the track.  And we should be able to implement behavior so that when you hold Shift, that same row of 8 encoders is mapped to the BoB parameters of the "currently selected" device on the track. Then, for example, Alt + Channel Strip could "flip the switch (Shift) default" where Alt + Channel Strip gives you the BoB parameters of the selected device by default and Shift gives you the "banks" of devices on the track.  You can sort of think of the LCD display separately from the encoder mapping in that respect. Currently in channel strip mode, those "device bank" encoders are "not mapped" for midi feedback.  The script just interprets encoder button clicks  to jump to device mode of that device.  (which actually consists of two steps, switch the selected device to that device, then switch to device mode and show the selected device's parameter page). So in Channel Strip mode those "device bank" encoder knob twists could be "mapped" for midi feedback to the BoB parameters of the selected device, but the script/LCD only shows the "parameter value" and/or responds to knob-twist events if/when Shift is pressed.

**Reply by markusschloesser** [reply]
> > If you put parameters only in the BoB bank then they will not show up
> 
> Not sure what you mean, but it may just be the way you tried to implement. Theoretically, we should be able to map "device parameters" to encoders in any order we want. As long as there is at least 1 parameter to map, it could be mapped to any single encoder up through all 32 encoders.

this refers to modifying the default devices.py by ableton (i also use github as notepad to jot things down, this was more a reminder of a quote I saw in the thread)
If you modify it, and only have that parameter in Bob, then it won't show up at all. 
Devices.py also where you could remap your eq8 Param order 😉

> 
> This seems like a great idea for an Alt + Channel Strip mode, and/or what you see when you hold Shift in Channel mode. I mean regular "channel strip" mode now shows the "channel banks" of devices (names) loaded on the track so you can button-click jump to device mode of that device regardless which device is "currently selected" on the track. And we should be able to implement behavior so that when you hold Shift, that same row of 8 encoders is mapped to the BoB parameters of the "currently selected" device on the track. Then, for example, Alt + Channel Strip could "flip the switch (Shift) default" where Alt + Channel Strip gives you the BoB parameters of the selected device by default and Shift gives you the "banks" of devices on the track. You can sort of think of the LCD display separately from the encoder mapping in that respect. Currently in channel strip mode, those "device bank" encoders are "not mapped" for midi feedback. The script just interprets encoder button clicks to jump to device mode of that device. (which actually consists of two steps, switch the selected device to that device, then switch to device mode and show the selected device's parameter page). So in Channel Strip mode those "device bank" encoder knob twists could be "mapped" for midi feedback to the BoB parameters of the selected device, but the script/LCD only shows the "parameter value" and/or responds to knob-twist events if/when Shift is pressed.

Fantastic idea!
